Update to 1.21.8, prepare for release

ActuallyTheOwner opened the pull request:
I rewrote the code for mob spawners and introduced a game rule "do_peaceful_spawns" that defaults to false
when turned on, this allows mob spawners in peaceful
There is now a dependency of the fabric API.

withers still do not spawn with their statue

ActuallyTheOwner committed:
rework SplashExtraMixin and update to 1.21.7

Idea by ioblackshaw

ActuallyTheOwner committed:
Fix spawning via spawners for most mobs with gamerule

Phantoms are still bugged and need a fix
Withers do not spawn with 3 soul sand and 3 wither skulls

ActuallyTheOwner committed:
Fix Silverfish

cputnam-a11y commented on src/main/java/com/skycatdev/nopeacefuldespawn/mixin/AllowInPeacefulMixin.java:
```diff
@@ -2,16 +2,35 @@
         return false;
     }
+
+    @WrapMethod(method = "canSpawnInDark")
```
this should be a WrapOperation on `world.getDifficulty() != Difficulty.PEACEFUL`

cputnam-a11y replied:
or a ModifyExpressionValue. either would work

ActuallyTheOwner committed:
rename do_peaceful_spawns to doPeacefulMobSpawns

the gamerule did not match vanilla's name scheme

cputnam-a11y commented on src/main/java/com/skycatdev/nopeacefuldespawn/mixin/AllowInPeacefulMixinSlimer.java:
```diff
@@ -0,0 +1,53 @@
+        return false;
+    }
+
+    @WrapMethod(method = "canSpawn")
```
this should be a WrapOperation or ModifyExpressionValue on `world.getDifficulty() != Difficulty.PEACEFUL`

cputnam-a11y commented on src/main/java/com/skycatdev/nopeacefuldespawn/mixin/AllowInPeacefulMixinSilverFishFix.java:
```diff
@@ -0,0 +1,41 @@
+
+@Mixin(InfestedBlock.class)
+public abstract class AllowInPeacefulMixinSilverFishFix {
+    @Inject(method = "onStacksDropped", at = @At("HEAD"), cancellable = true)
```
this could probably be a WrapOperation on `world.spawnEntity(silverfishEntity);` in `spawnSilverfish`. the only reason is because of how much code is copied, so it's closer to a matter of taste than the others

ActuallyTheOwner committed:
Fix phantoms, Update license, update to 1.21.8, prepare release!

Replaced placeholder license
added source code readme
Added clause against curseforge
Fix phantoms

cputnam-a11y commented on src/main/resources/fabric.mod.json:
```diff
@@ -3,14 +3,14 @@
+	"description": "Keeps monsters in Peaceful Mode with gamerule tweaks.",
 	"authors": [
-		"skycatminepokie"
+		"skycatminepokie, actuallytheowner"
```
```suggestion
		"skycatminepokie",
		"actuallytheowner"
```

ActuallyTheOwner replied:
I still want to credit myself for the mod as it has been rewritten a lot from the initial commit

cputnam-a11y replied:
Yes. I'm not suggesting you don't. I'm suggesting that you put yourself in a separate entry in the authors array so it's easier for mods like modmenu to render correctly

ActuallyTheOwner committed:
Remove redundant code

Lots of extra checks are not needed for the mod

cputnam-a11y commented on LICENSE:
The version of the licence should probably be changed to reflect the change in content and terms.

skycatminepokie replied:
In reality, I wrote that license cuz I was mad lol. @ActuallyTheOwner , what do you think of MIT? I'm fine with that, something else FOSS, or sticking with this.

ActuallyTheOwner committed:
change license version number and add hashes

skycatminepokie committed:
Change name casing

skycatminepokie committed:
Remove unecessary import

skycatminepokie committed:
List both as authors

This lists us both as authors instead of both of us as one author

skycatminepokie committed:
Rename NPaaceConfig

skycatminepokie committed:
Abstract SplashExtraMixin

skycatminepokie committed:
Clean up mixins
